concordium-consensus/tests/scheduler/SchedulerTests/SmartContracts/V1/Queries.hs

```diff
@@ -1,614 +1,724 @@
-                    },
-                  (SuccessWithSummary (deploymentCostCheck contractBalanceMissingContractSourceFile), emptySpec)
-                ),
-                ( TJSON
+                      metadata = makeDummyHeader accountAddress0 1 100_000,
+                      keys = [(0, [(0, keyPair0)])]
+                    },
+              taaAssertion = \result _ ->
+                return $ Helpers.assertSuccess result
+            },
+          Helpers.TransactionAndAssertion
+            { taaTransaction =
+                TJSON
                     { payload = InitContract 123 V1 contractBalanceMissingContractSourceFile "init_contract" "",
-                      metadata = makeDummyHeader alesAccount 2 100000,
-                      keys = [(0, [(0, alesKP)])]
-                    },
-                  (SuccessWithSummary (initializationCostCheck contractBalanceMissingContractSourceFile "init_contract"), emptySpec)
-                ),
-                ( TJSON
+                      metadata = makeDummyHeader accountAddress0 2 100_000,
+                      keys = [(0, [(0, keyPair0)])]
+                    },
+              taaAssertion = \result _ ->
+                return $ Helpers.assertSuccess result
+            },
+          Helpers.TransactionAndAssertion
+            { taaTransaction =
+                TJSON
                     { payload = Update 0 (Types.ContractAddress 0 0) "contract.query" parameters,
-                      metadata = makeDummyHeader alesAccount 3 100000,
-                      keys = [(0, [(0, alesKP)])]
+                      metadata = makeDummyHeader accountAddress0 3 100_000,
+                      keys = [(0, [(0, keyPair0)])]
                     },
-                  (SuccessWithSummary (successWithEventsCheck eventsCheck), emptySpec)
-                )
-            ]
-        }
-  where
+              taaAssertion = \result _ ->
+                -- Check the number of events:
+                -- - 1 event for a succesful update to the contract.
+                return $ Helpers.assertSuccessWhere (Helpers.assertNumberOfEvents 1) result
+            }
+        ]
     parameters = BSS.toShort $ runPut $ do
         putWord64le 1 -- Contract address index
         putWord64le 0 -- Contract address subindex
-    eventsCheck :: [Types.Event] -> Expectation
-    eventsCheck events = do
-        -- Check the number of events:
-        -- - 1 event for a succesful update to the contract.
-        eventsLengthCheck 1 events
 
 exchangeRatesSourceFile :: FilePath
 exchangeRatesSourceFile = "./testdata/contracts/v1/queries-exchange-rates.wasm"
 
-exchangeRatesTestCase :: TestCase PV5
-exchangeRatesTestCase =
-    TestCase
-        { tcName = "Query the exchange rates",
-          tcParameters = (defaultParams @PV5){tpInitialBlockState = initialBlockState},
-          tcTransactions =
-            [   ( TJSON
+exchangeRatesTestCase ::
+    forall pv.
+    Types.IsProtocolVersion pv =>
```

Comment:
Do you need this constraint, seeing that you just have `SProtocolVersion` already?

Reply:
Compiler complains if I remove it. `SProtocolVersion` does not have the constraint already, if that is what you mean.